js/build.js

```diff
@@ -1,371 +1,362 @@
+
+          if (str.length >= 4) {
+            if (str === _this.passcode) {
+              _this.savePasscodeOnPV(encryptPasscode(_this.passcode));
+              if (Fliplet.Env.get('platform') !== 'web' && _this.configuration.enable_touch_id) {
+                if (window.plugins.touchid) {
+                  window.plugins.touchid.isAvailable(
+                    function() {
+                      $lock.find('.state[data-state=verify]').removeClass('present').addClass('past');
+                      _this.calculateElHeight($lock.find('.state[data-state=touchID]'));
+                      $lock.find('.state[data-state=touchID]').removeClass('future').addClass('present');
+                      $(this).val('');
                     },
-                    function(msg) {
-                        // GA Track event
-                        Fliplet.Analytics.trackEvent({
-                          category: 'lock_screen',
-                          action: 'touchid_cancelled',
-                          nonInteraction: true
-                        });
+                    function() {
+                      $lock.find('.state[data-state=verify]').removeClass('present').addClass('past');
+                      _this.calculateElHeight($lock.find('.state[data-state=noTouchID]'));
+                      $lock.find('.state[data-state=noTouchID]').removeClass('future').addClass('present');
+                      $(this).val('');
                     }
-                );
-            },
-            savePasscodeOnPV: function(hashedPassCode) {
-                // GA Track event
-                Fliplet.Analytics.trackEvent({
-                  category: 'lock_screen',
-                  action: 'setup_success'
-                });
-
-                _this.passcodePV.hashedPassCode = hashedPassCode;
-                Fliplet.Security.Storage.update();
-            },
-            load_configuration: function(configuration) {
-                if (!configuration.has_reset) {
-                    $lock.find('.forgot-passcode').addClass("hidden");
+                  );
+                } else {
+                  $lock.find('.state[data-state=verify]').removeClass('present').addClass('past');
+                  _this.calculateElHeight($lock.find('.state[data-state=noTouchID]'));
+                  $lock.find('.state[data-state=noTouchID]').removeClass('future').addClass('present');
+                  $(this).val('');
                 }
+              } else {
+                $lock.find('.state[data-state=verify]').removeClass('present').addClass('past');
+                _this.calculateElHeight($lock.find('.state[data-state=noTouchID]'));
+                $lock.find('.state[data-state=noTouchID]').removeClass('future').addClass('present');
+                $(this).val('');
+              }
+            } else {
+              // GA Track event
+              Fliplet.Analytics.trackEvent({
+                category: 'lockScreen',
```

Comment:
Don't change the category for the tracked event name. This will affect our reporting.

```suggestion
                category: 'lock_screen',
```